apps/schema/src/components/TableEditModal.vue

```diff
@@ -1,156 +1,196 @@
+          );
+        }
+        return result;
       }
+      return undefined;
+    },
+    nameInvalid() {
+      if (
+        this.table.name === undefined ||
+        this.table.name.trim() === "" ||
+        this.table.name.search(/^[a-zA-Z0-9 _]*$/)
+      ) {
+        return "Name is required and can only contain 'azAZ_ '";
+      }
+      if (
+        this.value?.name !== this.table.name &&
+        ((this.schema.tables &&
+          this.schema.tables.filter(
+            (table) =>
+              table.name === this.table.name ||
+              (table.subclasses !== undefined &&
+                table.subclasses
+                  .map((subclass) => subclass.name)
+                  .includes(this.table.name))
+          ).length > 0) ||
+          (this.schema.ontologies &&
+            this.schema.ontologies.filter(
+              (ontology) => ontology.name === this.table.name
+            ).length > 0))
+      ) {
+        return "Name should be unique (no other table or ontology can have same name)";
+      }
+      return null;
+    },
+    subclassInvalid() {
+      return this.inheritOptions && this.table.inherit === undefined
+        ? "Extends is required in case of subclass"
+        : null;
+    },
+    isDisabled() {
+      return this.nameInvalid || this.subclassInvalid;
     },
-    action() {
-      if (this.tableDraft.command === 'CREATE') return `Create table`;
-      else return `Alter table ${this.table.name}`;
-    }
   },
   methods: {
-    executeCommand() {
-      this.loading = true;
-      this.graphqlError = null;
-      this.success = null;
-      request(
-        'graphql',
-        `mutation change($table:MolgenisTableInput){change(tables:[$table]){message}}`,
-        {
-          table: {
-            name: this.table.name,
-            description: this.table.description,
-            semantics: this.table.semantics
-          }
-        }
-      )
-        .then(() => {
-          if (this.table) {
-            this.success = `Table ${this.table.name} altered`;
-          } else {
-            this.success = `Table ${this.table.name} created`;
-          }
-          this.$emit('close');
-        })
-        .catch((error) => {
-          if (error.response.status === 403) {
-            this.graphqlError = 'Forbidden. Do you need to login?';
-            this.showLogin = true;
-          } else {
-            this.graphqlError = error.response.errors[0].message;
-            console.graphqlError(JSON.stringify(this.error));
-          }
-        });
-      this.loading = false;
-    }
-  },
-  watch: {
-    table: {
-      deep: true,
-      handler() {
-        this.created();
+    showModal() {
+      if (!this.value) {
+        this.table = {};
       }
-    }
+      this.show = true;
+    },
+    close() {
+      this.show = false;
+      this.$emit(this.operation, this.table);
+    },
+    cancel() {
+      //set
+      if (this.value) {
+        this.table = deepClonet(this.value);
```

Comment:
deepClonet should be deepClone?

Reply:
wow, I can't believe it still worked, I must have been badly testing. Maybe Eriks findings actually relate to this.